### DISCUSSION TITLE
A variable's time series generates, but the E3SM Diags area_mean_time_series plots don't generate

### DISCUSSION THREAD

**Discussion by forsyth2** [E3SM-Project/zppy#619] [Questions] [answered]
### Question criteria

- [X] I added a descriptive title here.
- [X] I searched the [zppy GitHub Discussions](https://github.com/E3SM-Project/zppy/discussions) to find a similar question and didn't find it.
- [X] I searched the [zppy documentation](https://e3sm-project.github.io/zppy).

### What is the deadline?

Not given

### Describe your question

From @evasinha 

Default EAM area mean time series plots, generated using zppy: https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.eva.sinha/20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3/e3sm_diags/atm_monthly_180x360_aave/model_vs_obs_2071-2090/viewer/area_mean_time_series/index.html

Cfg used: /home/ac.eva.sinha/E3SM-GCAM-GMD-2023/workflow/e3sm_diags/post.20240730.ZATM.chrysalis.cfg

Request: add more variables to the E3SM Diags area mean time series viewer. In particular, generate FLNT time series plots. The `ts` file was created for it but not plotted: /lcrc/group/e3sm/ac.eva.sinha/20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3/post/atm/180x360_aave/ts/monthly/20yr/FLNT_207101_209012.nc

### What machine were you running on?

Chrysalis

### Environment

Not given

### What command did you run?

```bash
zppy -c /home/ac.eva.sinha/E3SM-GCAM-GMD-2023/workflow/e3sm_diags/post.20240730.ZATM.chrysalis.cfg
```

### Copy your cfg file

```bash
[default]
input = /lcrc/group/e3sm/ac.eva.sinha/20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3
output = /lcrc/group/e3sm/ac.eva.sinha/20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3
case = 20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3
www = '/lcrc/group/e3sm/public_html/diagnostic_output/ac.eva.sinha'
partition = compute
environment_commands = "source /lcrc/soft/climate/e3sm-unified/load_latest_e3sm_unified_chrysalis.sh"

[climo]
active = True
years = "2071:2090:20",
walltime = "1:00:00"

  [[ atm_monthly_180x360_aave ]]
  input_subdir = "run"
  input_files = "eam.h0"
  mapping_file = /home/ac.zender/data/maps/map_ne30pg2_to_cmip6_180x360_aave.20200201.nc
  frequency = "monthly"

  [[ land_monthly_180x360_aave ]]
  input_subdir = "run"
  input_files = "elm.h0"
  mapping_file = "/lcrc/group/e3sm/public_html/diagnostics/maps/map_192x288_to_cmip6_180x360_aave.20240220.nc"
  frequency = "monthly"

[ts]
active = True
years = "2071:2090:20",
walltime = "1:00:00"

  [[ atm_monthly_180x360_aave ]]
  input_subdir = "run"
  input_files = "eam.h0"
  mapping_file = /home/ac.zender/data/maps/map_ne30pg2_to_cmip6_180x360_aave.20200201.nc
  frequency = "monthly"

  [[ land_monthly_180x360_aave ]]
  input_subdir = "run"
  input_files = "elm.h0"
  mapping_file = "/lcrc/group/e3sm/public_html/diagnostics/maps/map_192x288_to_cmip6_180x360_aave.20240220.nc"
  frequency = "monthly"
  vars = "SOILICE,SOILLIQ,SOILICE,QOVER,QRUNOFF,QINTR,QVEGE,QSOIL,QSOIL,QVEGT,TSOI,LAISHA,LAISUN,TOTLITC,CWDC,SOIL1C,SOIL2C,SOIL3C,WOOD_HARVESTC,LITFALL,LITTERC_LOSS,LITTERC_HR,CWDC_HR,TOTVEGC,NBP,GPP,AR,HR"
  extra_vars = "landfrac"

[e3sm_diags]
active = True
run_type = "model_vs_model"
walltime = "2:00:00"
years = "2071:2090:20",
ts_num_years = 20
ts_num_years_ref = 20
ref_start_yr = 2071
ref_end_yr = 2090
ref_final_yr = 2090
multiprocessing = True
num_workers = 8
short_name = 'With feedbacks'

  [[ atm_monthly_180x360_aave ]]
  ref_name = 20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3_without_feedbacks
  short_ref_name = 'Without feedbacks'
  diff_title = 'With feedbacks - Without feedbacks'
  grid = '180x360_aave'
  reference_data_path = '/lcrc/group/e3sm/ac.eva.sinha/20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3_without_feedbacks/post/atm/180x360_aave/clim'
  reference_data_path_ts = '/lcrc/group/e3sm/ac.eva.sinha/20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3_without_feedbacks/post/atm/180x360_aave/ts/monthly'
  sets = "lat_lon","area_mean_time_series",'polar',
  output_format_subplot = "pdf",

  [[ land_monthly_180x360_aave ]]
  ref_name = 20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3_without_feedbacks
  short_ref_name = 'Without feedbacks'
  diff_title = 'With feedbacks - Without feedbacks'
  grid = '180x360_aave'
  reference_data_path = '/lcrc/group/e3sm/ac.eva.sinha/20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3_without_feedbacks/post/lnd/180x360_aave/clim'
  reference_data_path_ts = '/lcrc/group/e3sm/ac.eva.sinha/20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3_without_feedbacks/post/lnd/180x360_aave/ts/monthly'
  sets = "lat_lon","area_mean_time_series"
  output_format_subplot = "pdf",
```

### What jobs are failing?

```bash
None
```

**Comment by chengzhuzhang** [comment, marked as answer]
@evasinha I can add `FLNT` to E3SM Diags, this require a code change of E3SM Diags. I will add this change. Alternatively, zppy already support global time-series output from this recent e3sm_unified deployment, @forsyth2 should be able to give instruction on add the global time-series task in the zppy config file.

**Reply by evasinha-pnnl** [reply]
@chengzhuzhang The  global time-series output from this recent e3sm_unified deployment should be sufficient.

I actually wanted to plot the net radiative flux at "top of atmosphere" and have just realized that is captured by the variable `RESTOM` which is included in the zppy plots.

But having the flexibility to generate global time-series output using the recent e3sm_unified will be helpful for other variables.

**Reply by forsyth2** [reply]
@evasinha Global time series example that plots a few atmospheric variables: https://github.com/E3SM-Project/zppy/blob/main/tests/integration/generated/test_min_case_global_time_series_custom_chrysalis.cfg

**Reply by evasinha-pnnl** [reply]
@forsyth2 Thank you very much for sharing instructions for generating global time series plots. I was able to generate the global time series plots. https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.eva.sinha/20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3/global_time_series/global_time_series_2071-2090_results/

**Comment by forsyth2** [comment]
## Initial Analysis 
### Errors
```
$ cd /lcrc/group/e3sm/ac.eva.sinha/20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3/post/scripts
$ grep FLNT *.o*
```
shows a lot of errors like:
```
RuntimeError: Neither does ERFari nor the variables in [('FSNT', 'FLNT', 'FSNT_d1', 'FLNT_d1')] exist in the file file:///lcrc/group/e3sm/ac.eva.sinha/20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3/post/scripts/tmp.568978.f0cW/climo_test/20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3_ANN_207101_209012_climo.nc
```

### What variables are being processed?
`vars` is not set for most of the cfg. That means it will default to https://github.com/E3SM-Project/zppy/blob/main/zppy/templates/default.ini#L64: `"FSNTOA,FLUT,FSNT,FLNT,FSNS,FLNS,SHFLX,QFLX,TAUX,TAUY,PRECC,PRECL,PRECSC,PRECSL,TS,TREFHT,CLDTOT,CLDHGH,CLDMED,CLDLOW,U"`

That is, by default, `FLNT` will be included. That's why the `ts > atm_monthly_180x360_aave` task is successful in producing `/lcrc/group/e3sm/ac.eva.sinha/20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3/post/atm/180x360_aave/ts/monthly/20yr/FLNT_207101_209012.nc`

So why isn't E3SM Diags also processing `FLNT`?

### E3SM Diags bash file

Based on this block:
```
{%- if ("enso_diags" in sets) or ("qbo" in sets) or ("area_mean_time_series" in sets) %}
{% if run_type == "model_vs_obs" %}
ts_dir_primary=ts
{% elif run_type == "model_vs_model" %}
ts_dir_primary=ts_test
{%- endif %}
# Create xml files for time series variables
ts_dir_source={{ output }}/post/atm/{{ grid }}/ts/monthly/{{ '%dyr' % (ts_num_years) }}
create_links_ts ${ts_dir_source} ${ts_dir_primary} ${Y1} ${Y2} 5
{% if run_type == "model_vs_model" %}
ts_dir_source={{ reference_data_path_ts }}/{{ ts_num_years_ref }}yr
ts_dir_ref=ts_ref
create_links_ts ${ts_dir_source} ${ts_dir_ref} ${ref_Y1} ${ref_Y2} 6
{%- endif %}
{%- endif %}
```
we know we're running
```
create_links_ts ${ts_dir_source} ${ts_dir_ref} ${ref_Y1} ${ref_Y2} 6
```
=>
```
create_links_ts {{ reference_data_path_ts }}/{{ ts_num_years_ref }}yr ts_ref ${ref_Y1} ${ref_Y2} 6
```
=>
```

create_links_ts /lcrc/group/e3sm/ac.eva.sinha/20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3_without_feedbacks/post/atm/180x360_aave/ts/monthly/{{ ts_num_years_ref }}yr ts_ref ${ref_Y1} ${ref_Y2} 6
```

Remember, we know the data exists at `/lcrc/group/e3sm/ac.eva.sinha/20240730_SSP245_ZATM_BGC_ne30pg2_f09_oEC60to30v3/post/atm/180x360_aave/ts/monthly/20yr/FLNT_207101_209012.nc`

So, it looks like the file linking is getting the correct directory. So, why isn't E3SM Diags picking up `FLNT`?

**Reply by forsyth2** [reply]
Ah, see @chengzhuzhang's reply https://github.com/E3SM-Project/zppy/discussions/619#discussioncomment-10491669

**Reply by chengzhuzhang** [reply]
`FLNT` is not supported as one of the default variables for `area_mean_time_series`...
here: https://github.com/E3SM-Project/e3sm_diags/blob/main/e3sm_diags/driver/default_diags/area_mean_time_series_model_vs_obs.cfg
